### DISCUSSION TITLE
HOGWARTS: House Also Changing on Previously Entered Cards with Each Sort Event

### DISCUSSION THREAD

**Discussion by thedigitalmenagerie** [nss-evening-cohort-14/evening-client-side#79] [Help Me!] [answered]
Honey-Rae Swan

Have you searched the other help tickets? 
Yes, I didn't see anything on this. 

Describe your issue:
Last night, I thought the sort button was working properly but each time 'sort' event happens, Hogwarts house is updating for previously sorted cards. 

List any error messages that appear: 
No error messages appear. 

What have you googled?
https://medium.com/@khaledhassan45/how-to-shuffle-an-array-in-javascript-6ca30d53f772
https://www.w3schools.com/js/js_random.asp
https://www.geeksforgeeks.org/for-each-loop-in-java/

What have you tried?
I have tried moving the array and the method within different parts of the code and it's current location is the closest I've made it. I know the function is right-- I am having issues understanding where it should go and, if it were to go elsewhere, what additional code would be needed to implement it. 

Copy/paste link TO THE FILE on your branch or in your repo:
https://github.com/thedigitalmenagerie/sorting-hat/tree/randomizer

**Comment by drteresavasquez** [comment]
@thedigitalmenagerie you need to also save the house that was assigned to the student in your object here: https://github.com/thedigitalmenagerie/sorting-hat/blob/randomizer/main.js#L38-L40

**Reply by thedigitalmenagerie** [reply]
That's what I thought and removed it because it gave me this error:

main.js:44 Uncaught ReferenceError: house is not defined
    at HTMLButtonElement.handleSortEvent (main.js:44)

**Reply by TrinityChristiana** [reply]
@thedigitalmenagerie How were you adding house to the object?

Option 1
```js
const nameIntoObj = {
    studentName,
    house
}
```
Option 2
```js
const nameIntoObj = {
    studentName,
    house: "House"
}
```

**Reply by thedigitalmenagerie** [reply]
Option 1-- Shorthand with a trailing comma.

**Reply by thedigitalmenagerie** [reply]
I tried option 2 (I remember you saying it didn't matter and that you could switch in the middle of the object's key value pairs). It didn't work that way either.

**Reply by thedigitalmenagerie** [reply]
Thank you, I kept trying to define it similarly to studentName above. Why? There is NO reason for house to use a .querySelector to insert something into the house portion of the card (what goes into house is worked out in the createStudentCards function) so, I used an empty array. That's why it wasn't working and I thought I had fixed it by deleting house within the object before seeing today that it had begun changing all the cards. 

Thank you guys!

**Comment by TrinityChristiana** [comment, marked as answer]
@thedigitalmenagerie `house` is not defined before you added it to your object. When you use shorthand the variable has to be defined before you can use it, like this:

```js
const variableName = "Test"

const newObj = {
    variableName,
}
```

Or option 2:

```js
const variableName = "Test"

const newObj = {
    item: variableName,
}
```

**Reply by thedigitalmenagerie** [reply]
I declared the array and the math method that sorts through the array within the function that was creating the cards, as opposed to the function that was handling the sort event, which was causing the event to run each time a card was created on every card, including previously entered cards. Moving this array and its math method to the function that handled the sort event pushes the information (just as it did with name) only to the new cards and does not alter previous cards.
